tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,268 @@
+            }
+
+            val run2 = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run2) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.logs.get.mkString(" ") should include(s"hello $testString")
+            }
+    }
+
+    it should "fail to invoke an action with an empty file" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "empty"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("empty.js")))
+            }
+            val run = wsk.action.invoke(name)
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "action developer error"
+                    activation.response.result shouldBe Some(JsObject("error" -> "Missing main/no code to execute.".toJson))
+            }
+    }
+
+    it should "create an action with an empty file" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "empty"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("empty.js")))
+            }
+            val rr = wsk.action.get(name)
+            wsk.parseJsonString(rr.stdout).getFieldPath("exec", "code") shouldBe Some(JsString(""))
+    }
+
+    it should "blocking invoke nested blocking actions" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "nestedBlockingAction"
+            val child = "wc"
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/wcbin.js")))
+            }
+            assetHelper.withCleaner(wsk.action, child) {
+                (action, _) => action.create(child, Some(TestUtils.getCatalogFilename("samples/wc.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson), blocking = true)
+            val activation = wsk.parseJsonString(run.stdout).convertTo[CliActivation]
+
+            withClue(s"check failed for activation: $activation") {
+                val wordCount = testString.split(" ").length
+                activation.response.result.get shouldBe JsObject("binaryCount" -> s"${wordCount.toBinaryString} (base 2)".toJson)
+            }
+    }
+
+    it should "blocking invoke an asynchronous action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloAsync"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/helloAsync.js")))
```

Comment:
Since I am refactoring all the test cases to remove the catalog, could you change getCatalogFilename into getTestActionFilename? If the script is not available under tests/dat/actions, please add it.